File: PruebaDigitalware.Models/Producto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace PruebaDigitalware.Models
{
    public class Producto
    {
        public Guid Id { get; set; }
        public string Nombre { get; set; }
        public decimal Costo { get; set; }
        public decimal Valor { get; set; }
        public int  Cantidad { get; set; }
        [ForeignKey("ProductoId")]
        public ICollection<FacturaProducto> FacturaProductos { get; set; }
    }
}

File: DigitalwareFront/src/app/Models/producto.ts
import { FacturaProducto } from './factura-producto';

export class Producto {

  id: string;
  nombre: string;
  costo: number;
  valor: number;
  cantidad: number;
  cantidadPedido: number = 0;
  facturaProductos: FacturaProducto[]; 

}
